packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef } from 'react';
+import { Dialog, DialogDisclosure, useDialogState } from 'reakit';
+import PrimitiveDrawer from '../../Primitive/PrimitiveDrawer';
+
+import theme from './FloatingDrawer.scss';
+
+export type DrawerProps = {
+	disclosure?: React.ReactElement;
+	header?: React.ReactNode;
+	children: React.ReactNode;
+	footer?: React.ReactNode;
+	visible?: boolean;
+	onClose?: () => void;
+};
```

Comment:
```suggestion
type WithDisclosure = {
   	disclosure: React.ReactElement;
   	visible?: never;
};

type Controlled = {
   	disclosure?: never;
   	visible: boolean;
};


export type DrawerProps = {
	header?: React.ReactNode;
	children: React.ReactNode;
	footer?: React.ReactNode;
	onClose?: () => void;
} & (WithDisclosure | Controlled);
```

Current typings open the component to impossible states

Reply:
Done